pallets/nfts/src/lib.rs

```diff
@@ -164,6 +165,19 @@ decl_module! {
 
             Self::deposit_event(RawEvent::Sealed(id));
         }
+
+        /// Remove an NFT from the storage. This operation is irreversible which means
+        /// once the NFT is removed (burned) from the storage there is no way to
+        /// get it back.
+        /// Must be called by the owner of the NFT.
+        #[weight = T::WeightInfo::burn()]
+        fn burn(origin, id: T::NFTId) {
+            let who = ensure_signed(origin)?;
+            let data = Data::<T>::get(id);
+
+            ensure!(data.owner == who, Error::<T>::NotOwner);
+            let _result = <Self as NFTs>::burn(id);
+        }
```

Comment:
This function should trigger an event

Reply:
May mistake, this should be for the function in the trait, not the dispatchables. Just like what we did for the others

Reply:
For some reason I thought that there was a separate task for that on Jira, my bad :)

Reply:
It should now be as proposed.